core/src/main/java/hudson/security/BasicAuthenticationFilter.java

```diff
@@ -162,15 +163,20 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             path += '?'+q;
 
         // prepare a redirect
-        rsp.setStatus(HttpServletResponse.SC_MOVED_TEMPORARILY);
-        rsp.setHeader("Location",path);
+        prepareRedirect(rsp, path);
 
         // ... but first let the container authenticate this request
         RequestDispatcher d = servletContext.getRequestDispatcher("/j_security_check?j_username="+
             URLEncoder.encode(username,"UTF-8")+"&j_password="+URLEncoder.encode(password,"UTF-8"));
         d.include(req,rsp);
     }
 
+    @SuppressFBWarnings(value = "UNVALIDATED_REDIRECT", justification = "Redirect is validated as processed.")
```

Comment:
I don't understand this comment.

Reply:
I'll see if I can investigate and improve this when I get a chance.